lib/Minz/Request.php

```diff
@@ -229,39 +276,46 @@ public static function bad($msg, $url = array()) {
 	 *         $default si $_GET[$param] n'existe pas
 	 */
 	public static function fetchGET($param = false, $default = false) {
-		if ($param === false) {
+		if (false === $param) {
 			return $_GET;
-		} elseif (isset($_GET[$param])) {
+		}
+		if (isset($_GET[$param])) {
 			return $_GET[$param];
-		} else {
-			return $default;
 		}
+		return $default;
 	}
 
 	/**
 	 * Allows receiving POST data as application/json
 	 */
 	private static function initJSON() {
-		$contentType = isset($_SERVER['CONTENT_TYPE']) ? $_SERVER['CONTENT_TYPE'] : '';
-		if ($contentType == '') {	//PHP < 5.3.16
-			$contentType = isset($_SERVER['HTTP_CONTENT_TYPE']) ? $_SERVER['HTTP_CONTENT_TYPE'] : '';
+		if ('application/json' !== static::extractContentType()) {
+			return;
+		}
+		if ('' === $ORIGINAL_INPUT = file_get_contents('php://input', false, null, 0, 1048576)) {
+			return;
 		}
-		$contentType = strtolower(trim($contentType));
-		if ($contentType === 'application/json') {
-			$ORIGINAL_INPUT = file_get_contents('php://input', false, null, 0, 1048576);
-			if ($ORIGINAL_INPUT != '') {
-				$json = json_decode($ORIGINAL_INPUT, true);
-				if ($json != null) {
-					foreach ($json as $k => $v) {
-						if (!isset($_POST[$k])) {
-							$_POST[$k] = $v;
-						}
-					}
-				}
+		if (null === $json = json_decode($ORIGINAL_INPUT, true)) {
+			return;
+		}
+
+		foreach ($json as $k => $v) {
+			if (!isset($_POST[$k])) {
+				$_POST[$k] = $v;
 			}
 		}
 	}
 
+	/**
+	 * @return string
+	 */
+	private static function extractContentType() {
+		if (null === $contentType = static::getHeader('CONTENT_TYPE')) {
+			$contentType = static::getHeader('HTTP_CONTENT_TYPE'); // PHP < 5.3.16
```

Comment:
I'm not sure we still need this since we dropped the support of PHP 5.3 :)

Reply:
I've done that before we drop 5.3 support :)